### ISSUE TITLE
Pilot Email Required Error

### ISSUE THREAD

**Issue by maiconschelter** [fastlane/fastlane#11767]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

In always case when I use fastlane pilot, this error it was ocurred for me. I set in the properties of pilot the email from tester, and set the Name, Last Name, Email, Group in the CSV file, but not foun'd. I don't have problem with the signature of app, because app is correct published on the iTunes Store, only test flight testers not added.

**[09:41:57]: {"code"=>"The email is required.", "description"=>"Failed to submit build for review."}**

What does happen here? Who is this email was required? Into my testers.csv file have one unique line like this: Maicon,Schelter,schelter@ig.com.br,Mobile.

<details>
  <pre>[09:41:53]: Distributing new build to testers: 0.0.7 - 0.0.7
+---------------------+-------------------------------------------------------------------------------------+
|                                               Lane Context                                                |
+---------------------+-------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM    | ios                                                                                 |
| PLATFORM_NAME       | ios                                                                                 |
| LANE_NAME           | ios build                                                                           |
| VERSION_NUMBER      | 0.0.7                                                                               |
| BUILD_NUMBER        | 7                                                                                   |
| CERT_FILE_PATH      | /Users/ipm/Documents/Mobile/MangosApp/autobuild/T9GJFKLB3G.cer                      |
| CERT_CERTIFICATE_ID | T9GJFKLB3G                                                                          |
| SIGH_PROFILE_PATH   | /Users/ipm/Documents/Mobile/MangosApp/autobuild/AppStore_br.com.ipm.mobile.MangosA  |
|                     | pp.mobileprovision                                                                  |
| SIGH_PROFILE_PATHS  | ["/Users/ipm/Documents/Mobile/MangosApp/autobuild/AppStore_br.com.ipm.mobile.Mango  |
|                     | sApp.mobileprovision"]                                                              |
| SIGH_UDID           | c3038c57-b281-423e-9536-33272d0adeb2                                                |
| SIGH_UUID           | c3038c57-b281-423e-9536-33272d0adeb2                                                |
| SIGH_NAME           | br.com.ipm.mobile.MangosApp AppStore                                                |
| SIGH_PROFILE_TYPE   | app-store                                                                           |
| PRODUCE_APPLE_ID    |                                                                                     |
+---------------------+-------------------------------------------------------------------------------------+
[09:41:57]: {"code"=>"The email is required.", "description"=>"Failed to submit build for review."}

+------+-------------------------------------------------------+-------------+
|                              fastlane summary                              |
+------+-------------------------------------------------------+-------------+
| Step | Action                                                | Time (in s) |
+------+-------------------------------------------------------+-------------+
| 1    | default_platform                                      | 0           |
| 2    | increment_version_number                              | 0           |
| 3    | increment_build_number                                | 0           |
| 4    | update_info_plist                                     | 0           |
| 5    | (cd ../../ && node ./scripts/callIncrement.js)        | 0           |
| 6    | (cd ../../platforms/ios && agvtool what-version)      | 0           |
| 7    | (cd ../../platforms/ios && agvtool                    | 0           |
|      | what-marketing-version -te                            |             |
| 8    | cert                                                  | 7           |
| 9    | sigh                                                  | 7           |
| 10   | (cd ../.. && ionic cordova build ios --device --prod  | 89          |
|      | --releas                                              |             |
| 11   | produce                                               | 3           |
| 💥   | pilot                                                 | 515         |
+------+-------------------------------------------------------+-------------+

[09:41:57]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  Pilot Distribute can't set Email
    https://github.com/fastlane/fastlane/issues/9594 [closed] 4 💬
    30 Nov 2017

🔗  You can ⌘ + double-click on links to open them directly in your browser.
/Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/spaceship/lib/spaceship/test_flight/client.rb:290:in **`handle_response': [!] {"code"=>"The email is required.", "description"=>"Failed to submit build for review."} (Spaceship::UnexpectedResponse)**
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/spaceship/lib/spaceship/test_flight/client.rb:72:in `post_for_testflight_review'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/spaceship/lib/spaceship/test_flight/build.rb:204:in `submit_for_testflight_review!'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/pilot/lib/pilot/build_manager.rb:149:in `distribute_build'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/pilot/lib/pilot/build_manager.rb:88:in `distribute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/pilot/lib/pilot/build_manager.rb:52:in `upload'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:231:in `block in execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
	from Fastfile:112:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/commands_generator.rb:337:in `run'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/fastlane/lib/fastlane/cli_tools_distributor.rb:98:in `take_off'
	from /Library/Ruby/Gems/2.3.0/gems/fastlane-2.80.0/bin/fastlane:20:in `<top (required)>'
	from /usr/local/bin/fastlane:22:in `load'
	from /usr/local/bin/fastlane:22:in `<main>'
</pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.13.2                                                        |
| Ruby                        | 2.3.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.14.3 (Apple Git-98)                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.13.2 (17C205)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.2.7                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 9.2                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | pt_BR.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#

ENV["APP_NAME"] = "MangosApp"

UI.message "+------+------------------+-------------+"
UI.message "|      IPM Fast Lane iOs Compiler       |"
UI.message "+------+------------------+-------------+"
UI.message "| Created by: Maicon Schelter           |"
UI.message "|  Create in: 2018-31-01                |"
UI.message "+------+------------------+-------------+"

default_platform(:ios)

platform :ios do

  lane :build do |options|

    if options[:increment]
      # Increment the version number
      increment_version_number(xcodeproj: "../platforms/ios/" + ENV["APP_NAME"] + ".xcodeproj")
      # Increment the build number
      increment_build_number(xcodeproj: "../platforms/ios/" + ENV["APP_NAME"] + ".xcodeproj")
      # Update project files
      update_info_plist(
        xcodeproj: "../platforms/ios/" + ENV["APP_NAME"] + ".xcodeproj",
        plist_path: "../platforms/ios/" + ENV["APP_NAME"] + "/" + ENV["APP_NAME"] + "-Info.plist"
      )
      # Increment the iOnic build number
      sh "(cd ../../ && node ./scripts/callIncrement.js)"
      # Show project version
      sh "(cd ../../platforms/ios && agvtool what-version)"
      sh "(cd ../../platforms/ios && agvtool what-marketing-version -terse1)"
      build_number = `cd ../../platforms/ios;agvtool what-marketing-version -terse1`.strip
    end

    if options[:compile]
      # Create distribution cert
      cert

      # Create/fetch provisioning profile
      profile_uuid = sigh

      # Delete iOS platform
      FileUtils.rm_rf('../../platforms/ios/build.json')
      FileUtils.rm_rf('../../platforms/ios/build/*')

      # Creating build.json folder to be used for release signing
      build_json = {
        "ios": {
          "release": {
            "codeSignIdentity": "iPhone Distribution",
            "provisioningProfile": "#{profile_uuid}",
            "packageType": "app-store",
            "developmentTeam": ENV["FASTLANE_TEAM_ID"]
          }
        }
      }

      # Writing build.json to platforms/ios directory
      out_file = File.new("../../platforms/ios/build.json", "w")
      out_file.puts(JSON.generate(build_json))
      out_file.close

      # Build iOS release IPA
      sh "(cd ../.. && ionic cordova build ios --device --prod --release --buildConfig=./platforms/ios/build.json)"
    end

    # Get IPA path
    ipa_path = Dir.glob(File.join(Dir.pwd, "../../platforms/ios/build/device/*.ipa")).first
    ipa_path = File.absolute_path ipa_path
    UI.success "Successfully built IPA: #{ipa_path}"

    if options[:produce]
      # Create app in the iOS dev center
      produce(
        app_name: 'MangosApp',
        app_version: build_number,
        language: 'English',
        sku: 'MangosApp_001',
        skip_itc: true,
        enable_services: {
          wireless_accessory: "on",
          push_notification: "on",
          network_extension: "on"
        }
      )
    end

    if options[:deliver]
      # Create app infos in iTC
      deliver(
        ipa: ipa_path,
        skip_screenshots: false,
        skip_metadata: false,
        skip_binary_upload: true,
        automatic_release: false,
        submit_for_review: false,
        app_version: version_num,
        app_icon: Dir.pwd + "/appicon.png",
        force: true
      )
    end

    if options[:pilot]
      # Create app in the iOs test flight
      pilot(
        ipa: ipa_path,
        changelog: "New version is available",
        beta_app_description: "New version for tests is available",
        beta_app_feedback_email: "ipm@ipm.com.br",
        distribute_external: true,
        first_name: "Maicon",
        last_name: "Schelter",
        email: "schelter@ig.com.br",
        testers_file_path: "./testers.csv",
        groups: "Mobile"
      )
    end

    # Clean up build temp files
    clean_build_artifacts

  end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "br.com.ipm.mobile.MangosApp"
apple_id "ipm@ipm.com.br"
team_id "QY8XBNLMS5"
itc_team_id "Y5568CRJKD"

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.80.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 2.3.2        |
| CFPropertyList            | 2.3.6        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.4        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.60.0       |
| faraday                   | 0.14.0       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.1        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.2.1        |
| bundler                   | 1.16.1       |
| faraday_middleware        | 0.12.2       |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.6       |
| io-console                | 0.4.5        |
| libxml-ruby               | 2.9.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
</details>

*generated on:* **2018-02-02**
</details>

**Comment by maiconschelter** [comment]
A second error ocurred now, but I really don't know why.

**[13:51:40]: {"code"=>"The email must be in the format: contact@example.com.", "description"=>"Failed to submit build for review."}**

Sometimes the first error occurs, sometimes this error occurs. Any ideas about this problem?

**Comment by getaaron** [comment]
You need to set the email address in the "Beta App Review Information" of iTunes Connect.

- You can put your app id into `https://itunesconnect.apple.com/WebObjects/iTunesConnect.woa/ra/ng/app/1234567890/testflight?section=testinformation`), or
- you can pass the `beta_app_feedback_email` option to _pilot_.

**Comment by maiconschelter** [comment]
Perfect! Many thanks @getaaron for the support.

**Comment by chrisva** [comment]
I still get this also and I have set `beta_app_feedback_email: 'mymail@mail.com'` option to `pilot`. Message from pilot says: `failed to publish to iTunes Connect. Error: #<Spaceship::Client::UnexpectedResponse: {"code"=>"The email is required.", "description"=>"Failed to submit build for review."}>`

When I log into iTunes I can see the e-mail is there. So that's strange. But these fields are missing. Do they need to be set, or can I set it using Fastlane pilot options?

<img width="995" alt="screen shot 2018-02-08 at 16 34 01" src="https://user-images.githubusercontent.com/1721274/35981621-fe6a4c52-0ced-11e8-893d-7cbc132a9bd3.png">
